### PR TITLE
EVG-19316: Enable @graphql-eslint/naming-convention rule

### PR DESCRIPTION
**Pull request by minnakt**
EVG-19316

### Description
This PR enables the rule [`@graphql-eslint/naming-convention`](https://the-guild.dev/graphql/eslint/rules/naming-convention).

### CODE REVIEW

File: src/analytics/task/useAnnotationAnalytics.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@ import { useGetUserQuery } from "analytics/useGetUserQuery";
 import {
   BuildBaronQuery,
   BuildBaronQueryVariables,
-  GetAnnotationEventDataQuery,
-  GetAnnotationEventDataQueryVariables,
+  AnnotationEventDataQuery,
+  AnnotationEventDataQueryVariables,
 } from "gql/generated/types";
 import { GET_ANNOTATION_EVENT_DATA, GET_BUILD_BARON } from "gql/queries";
```
Comment (sophstad) [inline src/analytics/task/useAnnotationAnalytics.ts:12]: What do you think about removing all the `GET_`s from these exports? We could do it as part of a different ticket.

**Review by sophstad — APPROVED** [review]
Wahoo 😎

File: src/analytics/task/useAnnotationAnalytics.ts (hunk 1 above)
Comment (minnakt) [reply src/analytics/task/useAnnotationAnalytics.ts:12]: sure! opened [EVG-19815](https://jira.mongodb.org/browse/EVG-19815)

**Review by khelif96 — APPROVED** [review]
Thank you for doing this!

**Pull request merged**